### PR TITLE
Fix 'Wrong message " The "<option_name>" option from the configuration file will be ignored. "' (close #4234)

### LINKED ISSUE #4234: The "reporter" option from the configuration file will be ignored.
### What is your Test Scenario?
I am running a simple test to check if the html body is present. With that, I am using a reporter in `.testcaferc.json`
 
### What is the Current behavior?
The test runs successfully but I get this message on console :
`The "reporter" option from the configuration file will be ignored.`

### What is the Expected behaviour?
The `.testcaferc.json` file shall find the reporter and save report accordingly.
 
### What is your web application and your TestCafe test code?

`.testcaferc.json` part 

```
{
"reporter": [
    {
      "name": "spec"
    },
    {
      "name": "cucumber-json",
      "output": "mycooltest7-2019-9-4-15-59-46-report.json"
    }
  ]
}
```
[cucumber-json](https://www.npmjs.com/package/testcafe-reporter-cucumber-json) is installed in the version 3.0.1 but this issue is anyway independent of which reporter I am using.
Same error happens with this configuration of reporter

```
{
"reporter": [
    {
      "name": "spec"
    }
  ]
}
```

Your website URL (or attach your complete example):

<details>
<summary>Your complete test code (or attach your test files):</summary>

 
```js
 
```
</details>

<details>
<summary>Your complete configuration file (if any):</summary>

```
{
  "browsers": "chrome:headless",
  "skipJsErrors": true,
  "selectorTimeout": 10000,
  "pageLoadTimeout": 10000,
  "speed": 1,
  "reporter": [
    {
      "name": "spec"
    },
    {
      "name": "cucumber-json",
      "output": "mycooltest7-2019-9-4-15-59-46-report.json"
    }
  ],
  "takeScreenshotsOnFails": true,
  "videoPath": "videos",
  "videoOptions": {
    "singleFile": false,
    "failedOnly": true,
    "pathPattern": "${FILE_INDEX}.mp4"
  },
  "screenshotPath": "screenshots",
  "screenshotPathPattern": "${FILE_INDEX}.png",
  "concurrency": 1,
  "quarantineMode": true,
}
```
</details>

<details>
<summary>Your complete test report:</summary>

```
The "reporter" option from the configuration file will be ignored.
 Running tests in:
 - Chrome 76.0.3809 / Mac OS X 10.13.6
```
</details>

<details>
<summary>Screenshots:</summary>

```

```
</details>

### Steps to Reproduce:

1. Go to my website ...
3. Execute this command...
4. See the error...
 
### Your Environment details:
 
* testcafe version: 1.4.3                   
* node.js version: 12.6.0                    
* command-line arguments:  testcafe test.js 
* browser name and version: Chrome 76.0.3809  
* platform and version:   Mac OS X 10.13.6        
* other:

### PR DESCRIPTION
**Pull request by miherlosev**

### CODE REVIEW

File: test/server/runner-test.js [hunk 1]
```diff
@@ -977,12 +977,13 @@ describe('Runner', () => {
         });
     });
 
-    it('Should interpret the empty array of the arguments as the "undefined" value (only in CLI mode for "browsers" and "src" methods)', () => {
+    it('Should interpret the empty array of the arguments as the "undefined" value', () => {
```
Comment (AlexKamaev) [inline test/server/runner-test.js]: Please check the naming or the test

File: src/cli/argument-parser.ts [hunk 2]
```diff
@@ -105,106 +132,109 @@ export default class CLIArgumentParser {
...
             return false;
         }
 
         return true;
     }
 
-    async _parseFilteringOptions () {
+    public async _parseFilteringOptions (): Promise<void> {
         if (this.opts.testGrep)
-            this.opts.testGrep = getGrepOptions('--test-grep', this.opts.testGrep);
```
Comment (LavrovArtem) [inline src/cli/argument-parser.ts]: if ('testGrep' in this.opts)

https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-the-in-operator
```diff
- this.opts.testGrep = getGrepOptions('--test-grep', this.opts.testGrep as string);
+ this.opts.testGrep = getGrepOptions('--test-grep', this.opts.testGrep);
```

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Comment by miherlosev** [comment]
@testcafe-build-bot retest

**Pull request merged**